Enhancement/front end using api/451

jfranco-gri opened the pull request:
Closes #451 

Aside from getting the front end code to use the Django API, I also removed all the esQueryBuilder and Elasticsearch javascript code since that is handled by the Django API now. I also updated unit and E2E tests to reflect my changes and wrote new Django tests for the API code.

I also used django-cors-headers to handle the cross-origin problem so that will need to be installed from the requirements file.

The API url is still in the form /api/books/<searchQuery> so let me know if that needs to be changed.

And lastly, this contains a lot of changes, so it would helpful to have some others pull it down and test it before merging.

**Reviewers**
- [ ] @joshuago78 
- [ ] @adamcee

adamcee commented:
@joshuago78 Pushed a commit with develop merged and conflicts fixed. All tests pass except the E2E saved records test which is at the moment also failing on develop.

joshuago78 commented:
The Django views have hardcoded urls. These should be configurable using the setting.py file instead

jfranco-gri commented:
Removed the hardcoded urls from views.py and added Elasticsearch host and port variables to settings.py. I followed the convention @adamcee has been using (separate variables for host and port and then combining them) but let me know if you would prefer just a global Elasticsearch address setting in settings.py instead, like so:

`ELASTICSEARCH_ADDRESS = 'local.portal.dev:9200'`
